cmd/dev/app/rule_type/rttst.go

```diff
@@ -148,38 +149,30 @@ func testCmdRun(cmd *cobra.Command, _ []string) error {
 
 	// Disable actions
 	off := "off"
-	p.Alert = &off
+	profile.Alert = &off
 
-	rules, err := engine.GetRulesFromProfileOfType(p, rt)
+	rules, err := engine.GetRulesFromProfileOfType(profile, ruletype)
 	if err != nil {
 		return fmt.Errorf("error getting relevant fragment: %w", err)
 	}
 
-	// TODO: Read this from a providers file instead so we can make it pluggable
-	eng, err := engine.NewRuleTypeEngine(context.Background(), p, rt, providers.NewProviderBuilder(
-		&db.Provider{
-			Name:    "test",
-			Version: "v1",
-			Implements: []db.ProviderType{
-				"rest",
-				"repo-lister",
-				"git",
-				"github",
-			},
-			Definition: json.RawMessage(`{
-				"github-app": {}
-			}`),
-		},
-		sql.NullString{},
-		false,
+	// TODO: Whenever we add more Provider classes, we will need to rethink this
+	client, err := clients.NewGitHubAppProvider(
+		&minderv1.GitHubAppProviderConfig{},
+		&serverconfig.GitHubAppConfig{AppName: "test"},
+		&ratecache.NoopRestClientCache{},
 		credentials.NewGitHubTokenCredential(token),
-		&serverconfig.ProviderConfig{
-			GitHubApp: &serverconfig.GitHubAppConfig{
-				AppName: "test",
-			},
-		},
-		nil, // this is unused here
-	))
+		nil,
+		clients.NewGitHubClientFactory(telemetry.NewNoopMetrics()),
+		false,
+	)
```

Comment:
This probably blocks https://github.com/stacklok/minder/pull/2983 from moving forward. What's the alternative? Seems like this locks us into a one-provider rule type engine for now which is not ideal.

Reply:
For posterity, we chatted about this on a call. I think we need to revisit this later - there are a bunch of things going on which may change what the ideal solution would be.